runhouse/main.py

```diff
@@ -666,6 +351,182 @@ def cluster_list(
+
+        try:
+            if current_cluster._http_client:
+                current_cluster._http_client.check_server()
+        except requests.exceptions.ConnectionError:
+            print_bring_cluster_up_msg(
+                cluster_name=cluster_name, msg_prefix="Can't get live cluster logs"
+            )
+            bring_logs_from_den = True
+
+    elif not cluster_or_local:
+        # todo: make func as the print_bring_cluster_up_msg
+        console.print(
+            "Can't get live cluster logs: \N{smiling face with horns} Runhouse Daemon is not running... \N{No Entry} \N{Runner}. "
+            "Start it with `runhouse restart` or specify a remote "
+            "cluster to poll with `runhouse status <cluster_name>`."
+        )
+        bring_logs_from_den = True
+
+    else:
+        # we are inside the cluster
+        current_cluster = cluster_or_local  # cluster_or_local = rh.here
+
+    try:
+        if bring_logs_from_den:
+            console.print("Getting the last chunk of logs saved in Den.")
+            get_logs_den_resp = current_cluster._get_saved_logs_from_den()
+            if get_logs_den_resp.status_code != 200:
+                console.print("Failed to get logs from den.")
+                return
+            logs = "\n".join(get_logs_den_resp.json().get("data").get("logs_text"))
+        else:
+            logs = current_cluster.logs(tail)
+
+    except requests.exceptions.ConnectionError:
+        console.print("Can't get live cluster logs, ")
+        console.print(
+            "\N{smiling face with horns} Runhouse Daemon is not running... \N{No Entry} \N{Runner}"
+        )
+        return
+
+    console.print(f"[reset][italic cyan]{logs}")
+
+
+@cluster_app.command("up")
+def cluster_up(
+    cluster_name: str = typer.Argument(
+        None,
+        help="The name of cluster to bring up.",
+    )
+):
+    """Bring up the cluster if it is not up. No-op if cluster is already up.
+    This only applies to on-demand clusters, and has no effect on self-managed clusters.
+
+    Example:
+        .. code-block:: console:
+
+            $ runhouse cluster up /sashab/rh-basic-cpu
+
+    """
+    try:
+        current_cluster = rh.cluster(name=cluster_name)
+        current_cluster.up_if_not()
+    except ValueError as e:
+        if str(e) == f"Resource {cluster_name} not found":
+            console.print("Cluster is not saved in den.")
+        else:
+            console.print(f"Failed to bring up the cluster: {e}")
+    except Exception as e:
+        console.print(f"Failed to bring up the cluster: {e}")
+
+
+@cluster_app.command("down")
+def cluster_down(
+    cluster_name: str = typer.Argument(
+        None,
+        help="The name of cluster to terminate.",
+    )
+):
+    """Terminate cluster if it is not down. No-op if cluster is already down.
+    This only applies to on-demand clusters, and has no effect on self-managed clusters.
+
+    Example:
+        .. code-block:: console:
```

Comment:
i dont think sphinx style code blocks are supported here - its not rending well in the CLI.. prob ok to just include the raw example snippet